projects/observability/src/shared/components/explore-query-editor/order-by/explore-query-order-by-editor.component.ts

```diff
@@ -0,0 +1,239 @@
+import { ChangeDetectionStrategy, Component, EventEmitter, Input, OnChanges, Output } from '@angular/core';
+import { TypedSimpleChanges } from '@hypertrace/common';
+import { SelectOption } from '@hypertrace/components';
+import { combineLatest, EMPTY, Observable, of, ReplaySubject, Subject } from 'rxjs';
+import { map, switchMap, take } from 'rxjs/operators';
+import { AttributeMetadata } from '../../../graphql/model/metadata/attribute-metadata';
+import { getAggregationDisplayName, MetricAggregationType } from '../../../graphql/model/metrics/metric-aggregation';
+import { GraphQlSortDirection } from '../../../graphql/model/schema/sort/graphql-sort-direction';
+import { TraceType } from '../../../graphql/model/schema/trace';
+import { MetadataService } from '../../../services/metadata/metadata.service';
+import { GraphQlOrderBy } from '../explore-visualization-builder';
+@Component({
+  selector: 'ht-explore-query-order-by-editor',
+  styleUrls: ['./explore-query-order-by-editor.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <div class="order-by-container" *ngIf="this.orderByExpression">
+      <span class="order-by-main-label"> Order by </span>
+      <div class="fields-container">
+        <div class="order-by-input-container" *ngIf="this.metricOptions$ | async as metricOptions">
```

Comment:
please use our async constructs like letAsync and loadAsync, rather than ngIf for side effects. I don't typically review PRs any more, but if this has become common, please push back on it - there's a reason we introduced those other directives.